Client/Application/AppDelegate.swift

```diff
@@ -662,6 +662,16 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UIViewControllerRestorati
 
     func application(_ application: UIApplication, continue userActivity: NSUserActivity, restorationHandler: @escaping ([Any]?) -> Void) -> Bool {
         if let url = userActivity.webpageURL {
+            
+            // Check for universal links and launch FxA Sign-in controller
+            let query = url.getQuery()
+            if query["deep_link"] != nil {
+                let strUrl = query["deep_link"]?.removingPercentEncoding
+                let loadUrl = URL(string: strUrl!)
+                browserViewController.switchToTabForURLOrOpen(loadUrl!, isPrivileged: true)
+                return true
+            }
```

Comment:
This code needs to be hardened. The `!` there means an explicit unwrap, which will crash the application if `strUrl` is invalid and `URL(string:)` returns nil.

```
if let deepLink = url.getQuery["deep_link"]?.removingPercentEncoding, let url = URL(string: deepLink) {
    browserViewController.switchToTabForURLOrOpen(loadUrl, isPrivileged: true)
    return true
}
```

Using explcit unwraps is always a red flag. It is almost never needed.